ui/app/AppLayouts/Profile/views/AdvancedView.qml

```diff
@@ -354,10 +354,26 @@ SettingsContentBase {
                 anchors.rightMargin: 0
                 text: qsTr("Debug")
                 isSwitch: true
-                switchChecked: root.advancedStore.isDebugEnabled
+                isEnabled: !root.advancedStore.isRuntimeLogLevelSet
+                switchChecked: root.advancedStore.isRuntimeLogLevelSet ? root.advancedStore.runtimeLogLevel !== "INFO" : root.advancedStore.isDebugEnabled
```

Comment:
This line makes me think that we're doing smth wrong 🤔 
Thinking this over.

Reply:
The state of the switch is determined by 2 possible inputs. Either : 
- The ENV variable or CLI was used, and the switch will have the state non-debug or debug. Here it should be fair to consider INFO as non-debug and the rest as Debug?
- The ENV variable or CLI wasn't used, and the switch will have the state non-debug or debug

Reply:
Here we set the switch value to `root.advancedStore.runtimeLogLevel !== "INFO"`. But we never changed the original `root.advancedStore.isDebugEnabled`. 

E.g. if I have `isDebugEnabled == false` and i run the app with `--log-level=TRUE`, I will see this swithc as `checked: true`. But I won't see the debug UI elements.

Reply:
This makes me think that maybe 2 separate controls is not that bad of an idea 🙄

Reply:
As discussed, I don't think we should have any spacial conditions on `switchChecked` here.
Instead, we should override the `isDebugEnabled` value in `src/app_service/service/node_configuration/service.nim` like this:

```nim
proc isDebugEnabled*(self: Service): bool =
  let logLevel = self.getLogLevel()
  if main_constants.runtimeLogLevelSet():
    logLevel = main_context.LOG_LEVEL
  return logLevel == $LogLevel.DEBUG || logLevel == $LogLevel.TRACE
```

Reply:
For the benefit of future readers, the comment addresses a divergence in perspective, not limited to the Settings feature but extending to other functionalities such as copying message IDs in the chat. Given my newness to the codebase, I initially believed that disabling the Setting when utilizing runtime environment variables or command-line parameters should be handled independently of disabling other functions, like copying message IDs. However, I eventually conceded that the former approach might introduce additional complexity to understanding the entire mechanism.

Reply:
> This makes me think that maybe 2 separate controls is not that bad of an idea 🙄

This requires discussion for sure... thanks for sharing!